src/pages/MemberProgress.jsx

```diff
@@ -1,39 +1,89 @@
+/* eslint-disable no-nested-ternary */
 import PropTypes from 'prop-types';
+import { Component } from 'react';
 import { Header } from '../components/Header/Header';
 import { Footer } from '../components/Footer/Footer';
 import { TableLine } from '../components/TableLine/TableLine';
 import { TableNav } from '../components/TableNav/TableNav';
+import EmptyData from '../components/EmptyData/EmptyData';
+import { Loader } from '../components/Loader/Loader';
 import { SideBar } from '../components/SideBar/SideBar';
 import { memberProgressFields } from '../services/fields-template';
+import { getMemberProgressById, getUserById } from '../services/services';
 
-export const MemberProgress = ({ userData, tasksData }) => (
-  <div className='memberProgress wrapper'>
-    <SideBar />
-    <div className='memberProgress__container'>
-      <Header userData={userData} />
-      <div className='memberProgress__container__title'>
-        <span className='memberProgress__container__title__text'>Ivan Progress:</span>
-      </div>
-      <div className='memberProgress__container__nav'>
-        <TableNav tableNavigationFields={memberProgressFields} />
-      </div>
-      <div className='memberProgress__container__table'>
-        {tasksData.map((item, index) => (
-          <TableLine
-            key={item.name + item.startDate}
-            number={index + 1}
-            name={item.name}
-            note={<div className='note__text'>{item.description}</div>}
-            date={item.startDate}
-          />
-        ))}
+export class MemberProgress extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tasksData: [],
+      isDataEmpty: false,
+      isDataLoad: false,
+      selectedUser: {},
+    };
+  }
+
+  async componentDidMount() {
+    const {
+      computedMatch: {
+        params: { userId },
+      },
+    } = this.props;
+
+    const tasksData = await getMemberProgressById(userId);
+    const selectedUser = await getUserById(userId);
+    if (tasksData) {
+      this.setState({ tasksData, isDataEmpty: tasksData[0], isDataLoad: true });
```

Comment:
use callback version of setState

Reply:
thx, I forgot about this method.
```suggestion
    this.setState(() => {
        const isDataEmpty = tasksData[0];
        const isDataLoad = true;

        return { tasksData, isDataEmpty, isDataLoad };
      });
    }
```